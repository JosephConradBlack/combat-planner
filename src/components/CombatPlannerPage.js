import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import { Container, Segment, Divider } from 'semantic-ui-react';

import Banner from './Banner';
import FighterList from './FighterList';
import ButtonGroup from './ButtonGroup';

import FighterEditableFormContainer from '../containers/FighterEditableFormContainer';

const StyledSegment = styled(Segment)`
  &&& {
    background-color: white;
  }
`;

const StyledContainer = styled(Container)`
  &&& {
    margin-top: 40px;
    margin-bottom: 40px;
  }
`;

const CombatPlannerPage = ({
  items,
  onFighterUpdate,
  onClearFighters,
  onNextFighter
}) => (
  <div>
    <Banner />
    <StyledSegment basic>
      <StyledContainer>
        <ButtonGroup
          items={items}
          onClearFighters={onClearFighters}
          onNextFighter={onNextFighter}
        />
        <Divider />
        <FighterList items={items} onFighterUpdate={onFighterUpdate} />
        <FighterEditableFormContainer />
      </StyledContainer>
    </StyledSegment>
  </div>
);

CombatPlannerPage.propTypes = {
  items: PropTypes.array,
  onClearFighters: PropTypes.func,
  onNextFighter: PropTypes.func
};

export default CombatPlannerPage;
