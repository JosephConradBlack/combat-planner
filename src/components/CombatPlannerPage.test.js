import React from 'react';

import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import CombatPlannerPage from './CombatPlannerPage';

describe('CombatPlannerPage', () => {
  it('renders ', () => {
    const component = shallow(<CombatPlannerPage />);

    expect(toJson(component)).toMatchSnapshot();
  });
});
