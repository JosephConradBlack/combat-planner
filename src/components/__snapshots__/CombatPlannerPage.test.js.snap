// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CombatPlannerPage renders  1`] = `
<div>
  <Banner />
  <Styled(Segment)
    basic={true}
  >
    <Styled(Container)>
      <ButtonGroup />
      <Divider />
      <FighterList
        items={Array []}
      />
      <Connect(FighterEditableForm) />
    </Styled(Container)>
  </Styled(Segment)>
</div>
`;
